apps/managedidentity/managedidentity_test.go

```diff
@@ -4,43 +4,399 @@ package managedidentity
+	Resource    string `json:"resource"`
+	TokenType   string `json:"token_type"`
+	ClientID    string `json:"client_id"`
+}
+
+type ErrorResponse struct {
+	StatusCode    int    `json:"statusCode"`
+	Message       string `json:"message"`
+	CorrelationID string `json:"correlationId,omitempty"`
+}
+
+type fakeClient struct{}
+type errorClient struct {
+	errResponse ErrorResponse
+}
+
+func fakeMIClient(mangedIdentityId ID, options ...ClientOption) (Client, error) {
+	fakeClient, err := New(mangedIdentityId, options...)
 
 	if err != nil {
 		return Client{}, err
 	}
 
-	return client, nil
+	return fakeClient, nil
 }
 
-func TestManagedIdentity(t *testing.T) {
-	client, err := fakeClient(SystemAssigned())
+func (*fakeClient) CloseIdleConnections()  {}
+func (*errorClient) CloseIdleConnections() {}
 
-	if err != nil {
-		t.Fatal(err)
+func (*fakeClient) Do(req *http.Request) (*http.Response, error) {
+	w := http.Response{
+		StatusCode: http.StatusOK,
+		Body:       io.NopCloser(strings.NewReader(getSuccessfulResponse(resource))),
+		Header:     make(http.Header),
+	}
+	return &w, nil
+}
+
+func (e *errorClient) Do(req *http.Request) (*http.Response, error) {
+	w := http.Response{
+		StatusCode: e.errResponse.StatusCode,
+		Body:       io.NopCloser(strings.NewReader(makeResponseWithErrorData(e.errResponse))),
+		Header:     make(http.Header),
+	}
+	return &w, nil
+}
+
+func getSuccessfulResponse(resource string) string {
+	expiresOn := time.Now().Add(1 * time.Hour).Unix()
+	response := SuccessfulResponse{
+		AccessToken: "fakeToken",
+		ExpiresOn:   expiresOn,
+		Resource:    resource,
+		TokenType:   "Bearer",
+		ClientID:    "client_id",
+	}
+	jsonResponse, _ := json.Marshal(response)
+	return string(jsonResponse)
+}
+
+func makeResponseWithErrorData(errRsp ErrorResponse) string {
+	response := ErrorResponse{
+		StatusCode:    errRsp.StatusCode,
+		Message:       errRsp.Message,
+		CorrelationID: errRsp.CorrelationID,
+	}
+	jsonResponse, _ := json.Marshal(response)
+	return string(jsonResponse)
+}
+
+func getMsiErrorResponse() string {
+	response := ErrorResponse{
+		StatusCode:    500,
+		Message:       "An unexpected error occurred while fetching the AAD Token.",
+		CorrelationID: "7d0c9763-ff1d-4842-a3f3-6d49e64f4513",
 	}
+	jsonResponse, _ := json.Marshal(response)
+	return string(jsonResponse)
+}
 
-	_, err = client.AcquireToken(context.Background(), "scope", WithClaims("claim"))
+func getMsiErrorResponseNotFound() string {
+	response := ErrorResponse{
+		StatusCode:    500,
+		Message:       "An unexpected error occurred while fetching the AAD Token.",
+		CorrelationID: "7d0c9763-ff1d-4842-a3f3-6d49e64f4513",
+	}
+	jsonResponse, _ := json.Marshal(response)
+	return string(jsonResponse)
+}
 
-	if err == nil {
-		t.Errorf("TestManagedIdentity: unexpected nil error from TestManagedIdentity")
+func getMsiErrorResponseNoRetry() string {
+	response := ErrorResponse{
+		StatusCode:    123,
+		Message:       "Not one of the retryable error responses",
```

Comment:
You didn't implement retry policies so I don't see the point of this.

Reply:
Removed the method